vault/identity_store_oidc_provider_test.go

```diff
@@ -216,6 +216,60 @@ func TestOIDC_Path_OIDC_ProviderScope_List(t *testing.T) {
 	expectStrings(t, respListScopeAfterDelete.Data["keys"].([]string), expectedStrings)
 }
 
+// TestOIDC_Path_OIDC_ProviderScope_DeleteWithExistingProvider tests that a
+// Scope cannot be deleted when it is referenced by a provider
+func TestOIDC_Path_OIDC_ProviderScope_DeleteWithExistingProvider(t *testing.T) {
+	c, _, _ := TestCoreUnsealed(t)
+	ctx := namespace.RootContext(nil)
+	storage := &logical.InmemStorage{}
+
+	// Create a test scope "test-scope" -- should succeed
+	resp, err := c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/scope/test-scope",
+		Operation: logical.CreateOperation,
+		Data: map[string]interface{}{
+			"template":    `{"groups": "{{identity.entity.groups.names}}"}`,
+			"description": "my-description",
+		},
+		Storage: storage,
+	})
+	expectSuccess(t, resp, err)
+
+	// Create a test provider "test-provider"
+	c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/provider/test-provider",
+		Operation: logical.CreateOperation,
+		Data: map[string]interface{}{
+			"scopes": []string{"test-scope"},
+		},
+		Storage: storage,
+	})
+	expectSuccess(t, resp, err)
+
+	// Delete test-scope -- should fail
+	resp, err = c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/scope/test-scope",
+		Operation: logical.DeleteOperation,
+		Storage:   storage,
+	})
+	expectError(t, resp, err)
+	// validate error message
+	expectedStrings := map[string]interface{}{
+		"unable to delete scope \"test-scope\" because it is currently referenced by these providers: test-provider": true,
+	}
+	expectStrings(t, []string{resp.Data["error"].(string)}, expectedStrings)
+
+	// Read "test-scope" again and validate
+	resp, _ = c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/scope/test-scope",
+		Operation: logical.ReadOperation,
+		Storage:   storage,
+	})
+	if resp != nil {
+		t.Fatalf("expected nil but got resp: %#v", resp)
```

Comment:
we should get success here after the Read operation. Similar to [this test](https://github.com/hashicorp/vault/pull/12272/commits/f72b1d899b4c1d5f6d5fe6285401d18c1cb14b7c#diff-1585ec2bb6ce0b81bb0d3c1115ac3c43c73e326c41020aa1caf7026069918e55R759).